src/task/MonitorTask.ts

```diff
@@ -145,7 +145,18 @@ export default class MonitorTask
     const resubscribeForDownlink = this.context.videoDownlinkBandwidthPolicy.wantsResubscribe();
     needResubscribe = needResubscribe || resubscribeForDownlink;
     if (resubscribeForDownlink) {
-      this.context.videosToReceive = this.context.videoDownlinkBandwidthPolicy.chooseSubscriptions();
+      const videoSubscriptionIdSet = this.context.videoDownlinkBandwidthPolicy.chooseSubscriptions();
+      // Same logic as in `ReceiveVideoStreamIndexTask`, immediately truncating rather then truncating on subscribe
+      // avoids any issues with components (e.g. transceiver controller) along the way.
+      this.context.videosToReceive = videoSubscriptionIdSet.truncate(
```

Comment:
dont we only need to truncate if `videoSubscriptionIdSet.size() > this.context.videosToReceive.size()` ?

Reply:
I figure thats probably done internally by `truncate` anyways. i.e. doubt theres any performance improvement and this is a bit cleaner.